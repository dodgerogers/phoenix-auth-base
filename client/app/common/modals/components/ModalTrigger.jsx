import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Button } from 'semantic-ui-react';
import { showModal } from '../actions';
import { modalIds } from '../constants';


const ModalTrigger = (props) => {
  const Wrapper = props.wrapper || Button;

  return (
    <Wrapper onClick={props.showModal(props.id)}>
      {props.text}
    </Wrapper>
  )
}

ModalTrigger.propTypes = {
  text: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  showModal: PropTypes.func.isRequired,
};

ModalTrigger.defaultProps = {
  text: 'Open Modal',
}

export { ModalTrigger as PureComponent };
export default connect(null, { showModal })(ModalTrigger);
