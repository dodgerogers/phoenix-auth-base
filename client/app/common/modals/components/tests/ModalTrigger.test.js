import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';
import toJson from 'enzyme-to-json';
import { Button, Menu } from 'semantic-ui-react';
import ModalTrigger, { PureComponent } from '../ModalTrigger';


describe('ModalTrigger', () => {
  describe('PureComponent', () => {
    const fullProps = () => ({
      showModal: jest.fn(),
      text: 'Open my modal',
      id: 'my-modal-id',
    });

    const pureComponent = (props) => shallow(
      <PureComponent {...props} />, { lifecycleExperimental: true }
    );

    it('matches snapshot', () => {
      const component = pureComponent(fullProps());

      expect(toJson(component)).toMatchSnapshot();
    });

    it('matches snapshot with custom wrapping component', () => {
      const props = fullProps();
      props.wrapper = Menu.Item;
      const component = pureComponent(props);

      expect(toJson(component)).toMatchSnapshot();
    });

    it('onClick calls showModal with given ID', () => {
      const props = fullProps();
      const component = pureComponent(props);

      component.find(Button).simulate('click');

      expect(props.showModal).toHaveBeenCalledWith(props.id);
    });
  });
});
