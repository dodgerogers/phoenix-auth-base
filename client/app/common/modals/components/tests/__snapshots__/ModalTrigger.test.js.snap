// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalTrigger PureComponent matches snapshot 1`] = `
<Button
  as="button"
>
  Open my modal
</Button>
`;

exports[`ModalTrigger PureComponent matches snapshot with custom wrapping component 1`] = `
<MenuItem>
  Open my modal
</MenuItem>
`;
